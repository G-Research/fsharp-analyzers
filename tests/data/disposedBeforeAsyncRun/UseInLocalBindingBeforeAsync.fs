module M

open System
open System.IO

let asyncReturningFunc () =
    let localFunc () =
        // should warn
        use t = new System.IO.FileStream("", IO.FileMode.Open)
        t |> ignore
        ()
    localFunc ()
    async {
        return "hi"
    }
