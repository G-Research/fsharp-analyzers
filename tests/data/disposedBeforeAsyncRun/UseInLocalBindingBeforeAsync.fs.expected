GRA-DISPBEFOREASYNC-001 | Warning | (9,12 - 9,13) | Object is disposed before returned async is run | []
